core/tests/protractor/learnerFlow.js

```diff
@@ -0,0 +1,460 @@
+    learnerDashboardPage.navigateToInCompleteSection();
+    learnerDashboardPage.navigateToIncompleteExplorationsSection();
+    learnerDashboardPage.expectTitleOfExplorationSummaryTileToMatch(
+      explorationTitle);
+  };
+
+  var checkCompleteExplorationSection = function(explorationTitle) {
+    learnerDashboardPage.navigateToCompletedSection();
+    learnerDashboardPage.navigateToCompletedExplorationsSection();
+    learnerDashboardPage.expectTitleOfExplorationSummaryTileToMatch(
+      explorationTitle);
+  };
+
+  var checkIncompleteCollectionSection = function(collectionTitle) {
+    learnerDashboardPage.navigateToInCompleteSection();
+    learnerDashboardPage.navigateToIncompleteCollectionsSection();
+    learnerDashboardPage.expectTitleOfCollectionSummaryTileToMatch(
+      collectionTitle);
+  };
+
+  var checkCompleteCollectionSection = function(collectionTitle) {
+    learnerDashboardPage.navigateToCompletedSection();
+    learnerDashboardPage.navigateToCompletedCollectionsSection();
+    learnerDashboardPage.expectTitleOfCollectionSummaryTileToMatch(
+      collectionTitle);
+  };
+
+  beforeAll(function() {
+    adminPage = new AdminPage.AdminPage();
+    libraryPage = new LibraryPage.LibraryPage();
+    learnerDashboardPage = new LearnerDashboardPage.LearnerDashboardPage();
+
+    if (browser.isMobile) {
+      var ADM_VISITOR = 'admVisitor';
+      users.createAndLoginAdminUserMobile(
+        'admVisitor@learner.com', ADM_VISITOR);
+      // Load /explore/24
+      adminPage.reloadExploration('learner_flow_test.yaml');
+      // Load /explore/22
+      adminPage.reloadExploration('protractor_mobile_test_exploration.yaml');
+      // Load /collection/1
+      adminPage.reloadCollection(1);
+      users.logout();
+    } else {
+      collectionEditorPage = new CollectionEditorPage.CollectionEditorPage();
+      creatorDashboardPage = new CreatorDashboardPage.CreatorDashboardPage();
+      explorationEditorPage = new ExplorationEditorPage.ExplorationEditorPage();
+      explorationEditorMainTab = explorationEditorPage.getMainTab();
+      explorationEditorSettingsTab = explorationEditorPage.getSettingsTab();
+      explorationPlayerPage = new ExplorationPlayerPage.ExplorationPlayerPage();
+    }
+  });
+
+  it('visits the exploration player and plays the correct exploration',
+    function () {
+      if (!browser.isMobile) {
+        users.createAndLoginUser('expCreator@learnerDashboard.com',
+          'expCreator');
+        // Create an exploration named 'Exploration Player Test'.
+        workflow.createAndPublishExploration(
+          'Exploration Player Test',
+          'Astronomy',
+          'To test the exploration player',
+          'English'
+        );
+        users.logout();
+      }
+      var PLAYER_USERNAME = 'expPlayerDesktopAndMobile';
+      users.createAndLoginUser(
+        'expPlayerDesktopAndMobile@learnerFlow.com', PLAYER_USERNAME);
+      if (browser.isMobile) {
```

Comment:
Why does this part need to branch? Why can't we use lines 170-172 generically instead?

Reply:
Removed this branching as well as other such branching. Confined them to limited lines. Done.